docs/src/index.md

```diff
@@ -11,3 +11,38 @@ CurrentModule = ArrayInterface
 Modules = [ArrayInterface]
 ```
 
+## Static Traits
+
+The size along one or more dimensions of an array may be know at compile time. 
+`ArrayInterface.known_size` is useful for extracting this information from array types and `ArrayInterface.size` is useful for extracting this information from an instance of an array.
+For example:
+
+```julia
+julia> a = ones(3)';
+
+julia> ArrayInterface.size(a)
+(static(1), 3)
+
+julia> ArrayInterface.known_size(typeof(a))
+(1, nothing)
+
+```
+
+This is useful for dispatching on known information about the size of an array:
+```julia
+fxn(x) = _fxn(ArrayInterface.size(x), x)
+_fxn(sz::Tuple{StaticInt{S1},StaticInt{S2}}, x) where {S1,S2} = ...
+_fxn(sz::Tuple{StaticInt{3},StaticInt{3}}, x) = ...
+_fxn(sz::Tuple{Int,StaticInt{S2}}, x) where {S2} = ...
+_fxn(sz::Tuple{StaticInt{S1},Int}, x) where {S1} = ...
+_fxn(sz::Tuple{Int,Int}, x) = ...
+```
+
+Generally, `ArrayInterface.size` uses the return of `known_size` to form a static value for those dimensions with known length and only queries dimensions corresponding to `nothing`.
```

Comment:
If an array has some known dimensions and some unknown, will `ArrayInterface.size` force a dynamic dispatch?

Reply:
The only way dispatch is dynamic is if you try to enforce staticness like this:
```julia
fxn(x) = _fxn(Static.static(ArrayInterface.size(x)), x)
_fxn(sz::Tuple{StaticInt{S1},StaticInt{S2}}, x) where {S1,S2} = ...
```

Otherwise the compiler should know that something like `_fxn(sz::Tuple{StaticInt{S1},Int}, x) where {S1} = ...` is the correct method.